markets/storageadapter/dealpublisher.go

```diff
@@ -204,6 +204,26 @@ func (p *DealPublisher) processNewDeal(pdeal *pendingDeal) {
 		return
 	}
 
+	pdealPropCid, err := pdeal.deal.Proposal.Cid()
+	if err != nil {
+		log.Warn("failed to calculate proposal CID for new pending Deal with piece cid %s", pdeal.deal.Proposal.PieceCID)
+		return
+	}
+
+	// Sanity check that new deal isn't already in the queue
+	for _, pd := range p.pending {
+		pdPropCid, err := pd.deal.Proposal.Cid()
```

Comment:
This isn't fast, but shouldn't be perf critical. I also considered just comparing the bytes of the signature, but that feels hacky?

Opinions welcome.

Reply:
I agree, it's not performance critical, this approach LGTM 👍